src/finput.js

```diff
@@ -230,13 +231,22 @@ class Finput {
     this.setValue(this.element.value);
   }
   /**
-   * On focus of the input - Select all text
+   * On focusing IN of the input
+   * DEFAULT:  Select all text
    * @param {e} Focus event
    */
   onFocusin(e) {
-    console.debug('Focus IN event', e);
-    this.element.selectionStart = 0;
-    this.element.selectionEnd = this.element.value.length;
+    if(this.options.onFocusinCallback){
+      console.debug('Custom Focus IN event', e);
+      let selection = this.options.onFocusinCallback(e);
```

Comment:
need to check if selection is defined here, if undefined then skip the next 2 lines